src/ast/passes/semantic_analyser.cpp

```diff
@@ -920,7 +921,7 @@ void SemanticAnalyser::visit(Call &call)
     check_nargs(call, 0);
   } else if (call.func == "print") {
     check_assignment(call, false, false, false);
-    if (in_loop() && is_final_pass()) {
+    if (in_loop() && is_final_pass() && call.vargs->at(0)->is_map) {
```

Comment:
JC, what is this additional check for?

Reply:
`print()` comes in two forms: print an entire map or print a single value.

Only the first form is asynchronous, so the second is safe to use inside loops. The error message below was already assuming users were printing an entire map (1st form), so really this check had been wrong for a long time, but I just noticed it when I was testing for-loops.

Reply:
Ah gotcha. Thanks for the explanation.